include/swift/AST/DiagnosticsSema.def

```diff
@@ -71,6 +71,11 @@ ERROR(could_not_find_value_dynamic_member_corrected,none,
 ERROR(could_not_find_value_dynamic_member,none,
       "value of type %0 has no dynamic member %2 using key path from root type %1",
       (Type, Type, DeclNameRef))
+ERROR(cannot_infer_contextual_keypath_type_specify_root,none,
+      "cannot infer key path type from context; consider explicitly adding a root type", ())
+ERROR(cannot_infer_keypath_root_anykeypath_context,none,
+     "'AnyKeyPath' does not provide enough context for root type to be inferred; "
+     "consider explicitly adding a root type", ())
```

Comment:
@hborla What is better - `consider explicitly adding ...` or `consider adding ... explicitly`?

Reply:
What about `consider explicitly specifying a root type`? That's consistent with the language in other error messages prompting the user to add an explicit type somewhere

I haven't caught up on the other conversation in this PR, but is there a reason we aren't providing this as a fix-it to insert a placeholder for the root type?

Reply:
I like that wording better, thank you! And to answer your question - we do provide a fix-it to add a placeholder for root type.

Reply:
Right! I agree this feels better. Thanks @hborla :)

Reply:
Fixed :)